v2/pkg/engine/resolve/fetch.go

```diff
@@ -20,6 +22,42 @@ type Fetch interface {
 	FetchKind() FetchKind
 	Dependencies() FetchDependencies
 	DataSourceInfo() DataSourceInfo
+
+	GetBuffer() *bytes.Buffer
+	ReportResponseSize(out *bytes.Buffer)
+}
+
+// FetchBufferSizeCalculator calculates the right size for a buffer based on the previous 64 fetches
```

Comment:
How can we be right by checking the last 64 fetches that this number represents the average / quantile of the overall traffic? I think this heuristic is too simple and doesn't reflect real world scenarios. I think working with a pre sized / capped buffer brings similar improvements.